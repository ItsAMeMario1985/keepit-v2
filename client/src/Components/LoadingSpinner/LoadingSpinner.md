```js { "props": { "className": "test" } }
<LoadingSpinner />
```
