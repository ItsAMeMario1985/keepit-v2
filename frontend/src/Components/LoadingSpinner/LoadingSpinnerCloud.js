import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { CloudIcon } from '../Icons'

export default function LoadingSpinner({}) {
  return (
    <StyledLoadingWrapper>
      <CloudIcon fill="var(--color-primary)" width="40"></CloudIcon>
    </StyledLoadingWrapper>
  )
}

const StyledLoadingWrapper = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;

  svg {
    animation: heartbeat 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  }

  @keyframes heartbeat {
    0% {
      transform: scale(0.75);
    }
    20% {
      transform: scale(1);
    }
    40% {
      transform: scale(0.75);
    }
    60% {
      transform: scale(1);
    }
    80% {
      transform: scale(0.75);
    }
    100% {
      transform: scale(0.75);
    }
  }
`

const StyledLoadingDiv = styled.div``
