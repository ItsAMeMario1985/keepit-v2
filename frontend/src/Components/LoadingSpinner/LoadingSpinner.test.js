import { render } from '@testing-library/react'
import LoadingSpinner from './LoadingSpinner'
describe('LoadingSpinner', () => {
  it('renders correctly', () => {
    const { container } = render(<LoadingSpinner />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
