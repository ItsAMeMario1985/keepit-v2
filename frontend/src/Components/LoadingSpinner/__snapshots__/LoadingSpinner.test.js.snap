// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly 1`] = `
<div
  class="LoadingSpinner__StyledLoadingWrapper-vjl14b-0 WePaw"
>
  <div
    class="LoadingSpinner__StyledLoadingDiv-vjl14b-1 caFZkY"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
